biosimulators_utils/sedml/validation.py

```diff
@@ -1371,8 +1374,7 @@ def validate_data_generator_variables(variables, model_etrees=None, validate_tar
             models = get_models_referenced_by_task(variable.task)
             for model in models:
                 if model and model.language:
-                    model_changes = model.changes or list(filter(lambda change: change.model == model,
-                                                                 get_model_changes_for_task(variable.task)))
+                    model_changes = model.structural_changes
```

Comment:
Its best not to try to change the data structure because there are many downstream consequences.